### PR TITLE
switch to pyproject.toml

### PR DESCRIPTION
**Pull request by 180909**

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+exclude = static,.git,*migrations*,build,.tox,docs
+max-line-length = 119
+max-complexity = 7
```
Comment (greyli) [inline .flake8:4]: Always add an empty line at the end of the file.

**Review by greyli — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+"Issue Tracker" = "https://github.com/helloflask/bootstrap-flask/issues/"
+Funding = "https://opencollective.com/bootstrap-flask"
+Discussions = "https://github.com/helloflask/bootstrap-flask/discussions/"
+
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools.dynamic]
+version = { attr = "flask.__version__" }
```
Comment (greyli) [inline pyproject.toml:43]: Why use Flask's version? We could use an explicit version field.

File: .flake8 (hunk 1 above)
Comment (180909) [reply .flake8]: Done.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    'Framework :: Flask',
+    'Topic :: Internet :: WWW/HTTP :: Dynamic Content',
+    'Topic :: Software Development :: Libraries :: Python Modules',
+]
+dependencies = [
+    "Flask",
+    "WTForms",
+]
+dynamic = ["version"]
+
```
Comment (greyli) [inline pyproject.toml:29]: Based on the CI error output, adding a  `packages` or `py_modules` field is necessary.

**Review by greyli — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (180909) [reply pyproject.toml:26]: This is caused by tool.setuptools.dynamic .

**Review by greyli — APPROVED** [review]

**Comment by greyli** [comment]
Merged, thanks!

**Pull request merged**